pkg/ottl/ottlfuncs/func_is_match.go

```diff
@@ -32,11 +34,20 @@ func IsMatch[K any](target ottl.Getter[K], pattern string) (ottl.ExprFunc[K], er
 		if err != nil {
 			return nil, err
 		}
-		if val != nil {
-			if valStr, ok := val.(string); ok {
-				return compiledPattern.MatchString(valStr), nil
-			}
+
+		switch v := val.(type) {
+		case string:
+			return compiledPattern.MatchString(v), nil
+		case bool:
+			return compiledPattern.MatchString(strconv.FormatBool(v)), nil
+		case int64:
+			return compiledPattern.MatchString(strconv.FormatInt(v, 10)), nil
+		default:
```

Comment:
Should you have a case for `pcommon.Value`?

Reply:
Looking through our contexts we don't have any accessors that return a pcommon.Value, but I can add the case in anticipation of that changing in the future.

Reply:
Done